ding_periodical.module

```diff
@@ -135,12 +136,17 @@ function theme_ding_periodical_issues($variables) {
 
       // Check if the volume is reservable and add reservation button.
       if (!empty($issue_id) && ding_periodical_is_reservable($issues[$key])) {
-        $path = array($issue_id, $volume, $key);
+        $item_id = array($issue_id, $volume, $key);
+
+        foreach ($item_id as &$v) {
+          $v = rawurlencode($v);
```

Comment:
I'm not sure what should be moved to provider module, since the main issue is the hook_menu in ding_periodical that doesn't accept the periodical item id as it is expected.
And since it's decoding the argument from any provider, then it's true do encode them here for all providers as well.